File: irank_timer/tags/v1.2/iRank Timer/BlindLevel.m
//
//  BlindLevel.m
//  iRank Timer
//
//  Created by Luciano Stegun on 2013-09-22.
//  Copyright (c) 2013 Stegun.com. All rights reserved.
//

#import "BlindLevel.h"

@implementation BlindLevel
@synthesize levelIndex;
@synthesize levelNumber;
@synthesize smallBlind;
@synthesize bigBlind;
@synthesize ante;
@synthesize duration;
@synthesize seconds;
@synthesize elapsedTime;
@synthesize isBreak;

-(id)initWithLevelNumber:(int)pLevelNumber levelIndex:(int)pLevelIndex smallBlind:(int)pSmallBlind bigBlind:(int)pBigBlind ante:(int)pAnte duration:(int)pDuration {
    
    self = [super init];
    
    self.levelIndex  = pLevelIndex;
    self.levelNumber = pLevelNumber;
    self.smallBlind  = pSmallBlind;
    self.bigBlind    = pBigBlind;
    self.ante        = pAnte;
    self.duration    = pDuration;
    self.isBreak     = NO;
    self.seconds     = duration*60;
    
    return self;
}

-(id)initWithDuration:(int)pDuration isBreak:(int)pIsBreak levelIndex:(int)pLevelIndex {
    
    self = [super init];
    
    self.levelIndex  = pLevelIndex;
    self.levelNumber = 0;
    self.smallBlind  = 0;
    self.bigBlind    = 0;
    self.ante        = 0;
    self.duration    = pDuration;
    self.isBreak     = pIsBreak;
    self.seconds     = duration*60;
    
    return self;
}

-(id)copyWithZone:(NSZone *)zone {
    
    BlindLevel *copy = [[BlindLevel alloc] init];
    
    if( copy ){
        
        [copy setLevelIndex:  self.levelIndex];
        [copy setLevelNumber: self.levelNumber];
        [copy setSmallBlind:  self.smallBlind];
        [copy setBigBlind:    self.bigBlind];
        [copy setAnte:        self.ante];
        [copy setDuration:    self.duration];
        [copy setSeconds:     self.seconds];
        [copy setIsBreak:     self.isBreak];
        [copy setElapsedTime: self.elapsedTime];
    }
    
    return copy;
}

- (void)encodeWithCoder:(NSCoder *)encoder {
    
    [encoder encodeInt:levelIndex forKey:@"levelIndex"];
    [encoder encodeInt:levelNumber forKey:@"levelNumber"];
    [encoder encodeInt:smallBlind forKey:@"smallBlind"];
    [encoder encodeInt:bigBlind forKey:@"bigBlind"];
    [encoder encodeInt:ante forKey:@"ante"];
    [encoder encodeInt:duration forKey:@"duration"];
    [encoder encodeInt:seconds forKey:@"seconds"];
    [encoder encodeBool:isBreak forKey:@"isBreak"];
    [encoder encodeObject:elapsedTime forKey:@"elapsedTime"];
}

- (id)initWithCoder:(NSCoder *)decoder {
    
    self = [super init];
    
    [self setLevelIndex:  [decoder decodeIntForKey:@"levelIndex"]];
    [self setLevelNumber: [decoder decodeIntForKey:@"levelNumber"]];
    [self setSmallBlind:  [decoder decodeIntForKey:@"smallBlind"]];
    [self setBigBlind:    [decoder decodeIntForKey:@"bigBlind"]];
    [self setAnte:        [decoder decodeIntForKey:@"ante"]];
    [self setDuration:    [decoder decodeIntForKey:@"duration"]];
    [self setSeconds:     [decoder decodeIntForKey:@"seconds"]];
    [self setIsBreak:     [decoder decodeBoolForKey:@"isBreak"]];
    [self setElapsedTime: [decoder decodeObjectForKey:@"elapsedTime"]];
    
    return self;
}

-(void)setDuration:(int)pDuration {
    
    duration = pDuration;
    seconds  = pDuration*60;
}

-(NSString *)description {
    
    return [NSString stringWithFormat:@"#%i (%i) SB:%i BB:%i - %i (%i)", levelNumber, levelIndex, smallBlind, bigBlind, duration, isBreak];
}

@end

File: iRank Timer/BlindLevel.swift
//
//  BlindLevel.swift
//  iRank Timer
//
//  Created by Luciano Stegun on 05/02/15.
//  Copyright (c) 2015 Stegun.com. All rights reserved.
//

import Foundation

@objc(BlindLevel)
class BlindLevel : NSObject, NSCoding {
    
    var levelIndex : Int = 0;
    var levelNumber : Int = 0;
    var smallBlind : Int = 0;
    var bigBlind : Int = 0;
    var ante : Int = 0;
    var duration : Int = 0;
    var seconds : Int = 0;
    var isBreak : Bool = false;
    var elapsedTime : String = "00:00:00";
    
    override init(){
        
        self.levelIndex = 0;
    }

    override func copy() -> AnyObject {
        
        var blindLevel = BlindLevel();
        blindLevel.levelIndex  = self.levelIndex;
        blindLevel.levelNumber = self.levelNumber;
        blindLevel.smallBlind  = self.smallBlind;
        blindLevel.bigBlind    = self.bigBlind;
        blindLevel.ante        = self.ante;
        blindLevel.duration    = self.duration;
        blindLevel.seconds     = self.seconds;
        blindLevel.isBreak     = self.isBreak;
        blindLevel.elapsedTime = self.elapsedTime;
    
        return blindLevel;
    }
    
    required init(coder aDecoder: NSCoder) {
        
        self.levelIndex  = Int(aDecoder.decodeIntForKey("levelIndex"));
        self.levelNumber = Int(aDecoder.decodeIntForKey("levelNumber"));
        self.smallBlind  = Int(aDecoder.decodeIntForKey("smallBlind"));
        self.bigBlind    = Int(aDecoder.decodeIntForKey("bigBlind"));
        self.ante        = Int(aDecoder.decodeIntForKey("ante"));
        self.duration    = Int(aDecoder.decodeIntForKey("duration"));
        self.seconds     = Int(aDecoder.decodeIntForKey("seconds"));
        self.isBreak     = aDecoder.decodeBoolForKey("isBreak");
        self.elapsedTime = aDecoder.decodeObjectForKey("elapsedTime") as! String;
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        
        aCoder.encodeInt(Int32(self.levelIndex), forKey:"levelIndex");
        aCoder.encodeInt(Int32(self.levelNumber), forKey:"levelNumber");
        aCoder.encodeInt(Int32(self.smallBlind), forKey:"smallBlind");
        aCoder.encodeInt(Int32(self.bigBlind), forKey:"bigBlind");
        aCoder.encodeInt(Int32(self.ante), forKey:"ante");
        aCoder.encodeInt(Int32(self.duration), forKey:"duration");
        aCoder.encodeInt(Int32(self.seconds), forKey:"seconds");
        aCoder.encodeBool(self.isBreak, forKey:"isBreak");
        aCoder.encodeObject(self.elapsedTime, forKey:"elapsedTime");
    }
}